### PR TITLE
refactor: use github.com/natefinch/wrap

### PR DESCRIPTION
**Pull request by jpadilla**
## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Review by miparnisari — APPROVED** [review]

File: pkg/server/server.go [hunk 1]
```diff
@@ -372,6 +372,10 @@ func (s *Server) ListObjects(ctx context.Context, req *openfgav1.ListObjectsRequ
 	)
 	if err != nil {
 		if errors.Is(err, condition.ErrEvaluationFailed) {
+			var evalError *condition.EvaluationError
+			if errors.As(err, &evalError) {
```
Comment (jon-whit) [inline pkg/server/server.go:376]: Why is this indirection necessary? Just pipe it straight through `ValidationError` as is. No need for the error cast.

**Review by jon-whit — CHANGES_REQUESTED** [review]
Left one comment

File: pkg/server/server.go (hunk 1 above)
Comment (jpadilla) [reply pkg/server/server.go:376]: @jon-whit it boils down to this https://github.com/natefinch/wrap/blob/main/with.go#L108-L120

we'd like this:

```
{
	"code": "validation_error",
	"message": "failed to evaluate relationship condition 'condition2': context is missing parameters '[param1]'"
}
```

vs

```
{
	"code": "validation_error",
	"message": "failed to evaluate relationship condition: failed to evaluate relationship condition 'condition2': context is missing parameters '[param1]'"
}
```

Comment (jon-whit) [reply pkg/server/server.go:376]: Gotcha, thanks for the clarification.

**Review by jon-whit — APPROVED** [review]

**Pull request merged**